src/main/java/com/mazyavr/schedule/controller/UserController.java

```diff
@@ -0,0 +1,242 @@
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.api.services.calendar.model.Events;
+import com.mazyavr.schedule.dto.SimpleResponse;
+import com.mazyavr.schedule.dto.UserDto;
+import com.mazyavr.schedule.entity.TaskEntity;
+import com.mazyavr.schedule.entity.UserEntity;
+import com.mazyavr.schedule.repository.ProjectRepository;
+import com.mazyavr.schedule.repository.TaskRepository;
+import com.mazyavr.schedule.repository.UserRepository;
+import com.mazyavr.schedule.service.UserService;
+import java.security.GeneralSecurityException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder.In;
+import org.apache.tomcat.jni.Time;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Controller
+@RequestMapping("/user")
+@CrossOrigin(origins = "http://localhost:3000", allowCredentials = "true")
+public class UserController {
+
+  final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+
+  @Autowired
+  private UserRepository userRepository;
+  @Autowired
+  private ProjectRepository projectRepository;
+  @Autowired
+  private TaskRepository taskRepository;
+
+  private String token;
+  private Integer lifespan;
+
+  private Calendar initCalendar(long userId) {
+
+    Calendar userCalendar = null;
+
+    String userToken = userRepository.findById(userId).get().getToken();
+
+    if (userToken == null) {
+      throw new RuntimeException("Unable to initialize google calendar");
+    }
+
+    try {
+      var HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+      var credential = new GoogleCredential().setAccessToken(userToken);
+
+      userCalendar = new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
+          .setApplicationName("DownScheduler")
+          .build();
+    } catch (IOException | GeneralSecurityException e) {
+
+    }
+
+    if (userCalendar == null) {
+      throw new RuntimeException("Unable to initialize google calendar");
+    }
+    return userCalendar;
```

Comment:
Можно чуть проще написать

```suggestion
    try {
      var HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
      var credential = new GoogleCredential().setAccessToken(userToken);

      return new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
          .setApplicationName("DownScheduler")
          .build();
    } catch (IOException | GeneralSecurityException e) {
      throw new RuntimeException("Unable to initialize google calendar");
    }
```